gk/main.c

```diff
@@ -1188,12 +1384,27 @@ gk_proc(void *arg)
 	uint16_t rx_queue_back = instance->rx_queue_back;
 	uint16_t tx_queue_back = instance->tx_queue_back;
 
+	int num_buckets = rte_hash_get_num_buckets(
+		instance->ip_flow_hash_table);
+	/* Timeout for scanning one flow table bucket in ms. */
+	unsigned int bucket_scan_timeout_ms = round(
+		gk_conf->flow_table_scan_timeout_ms / num_buckets + 0.5);
+	uint64_t bucket_scan_timeout_cycles = round(
+		bucket_scan_timeout_ms * rte_get_tsc_hz() / 1000.);
+	uint32_t bucket_idx = 0;
+	uint32_t next = 0;
+	uint64_t request_timeout_cycles = gk_conf->request_timeout_sec *
+		rte_get_tsc_hz();
```

Comment:
1. Add `(uint64_t)` just before `gk_conf->request_timeout_sec` to guarantee that the product won't be truncated.

2. All uses of `gk_conf->request_timeout_sec` multiply it by `rte_get_tsc_hz()`. It's better to add a C function to be called in Lua to set a private field of `struct gk_config` instead of keeping computing this multiplication over and over again. Name suggestion for this field that would replace `request_timeout_sec`: `request_timeout_cycles`.